config/initializers/secure_headers.rb

```diff
@@ -59,12 +59,4 @@
   # }
 end
 
-SecureHeaders::Configuration.override(:saml) do |config|
-  provider_attributes = SERVICE_PROVIDERS.values
-
-  acs_urls = provider_attributes.map { |hash| hash['acs_url'] }.compact
-
-  whitelisted_domains = acs_urls.map { |url| url.split('//')[1].split('/')[0] }
-
-  whitelisted_domains.each { |domain| config.csp[:form_action] << domain }
-end
+SecureHeadersWhitelister.new.run
```

Comment:
yes. I have changed it in #972 to call only once at start up. Calling it multiple times is not an issue but at least now it is clearer that the controller does it, not the service class.